docker/config/config.go

```diff
@@ -55,15 +55,19 @@ func (c *Config) GetCredentials(registry string) (string, string, bool) {
 }
 
 func getAuthJSONString(username, password string) string {
-	if username == "_json_key" {
-		return fmt.Sprintf("%s:%s", username, password)
+	b, err := json.Marshal(types.AuthConfig{
+		Username: "_json_key",
+		Password: password,
+	})
+
+	// Because of the shape of the struct and inputs involved, this should never
+	// happen. We preserve the non error-propagating API for callers, but want
+	// some visibility into this that's better than simply swallowing the error.
+	if err != nil {
+		panic(err)
```

Comment:
Potentially a bit lazy here, but I didn't want to mess with your API's to much by changing the signature here.

Reply:
Cool, if I see this panic popping out anywhere, I would change, but anyway, this should never happen. :slightly_smiling_face: